src/aiProviders/continue/continueContextProvider.ts

```diff
@@ -0,0 +1,173 @@
+import * as vscode from "vscode";
+import { JobManager } from "../../config";
+import { JobInfo } from "../../connection/manager";
+import { SelfCodeNode } from "../../views/jobManager/selfCodes/nodes";
+import { findPossibleTables } from "../../chat/context";
+import {
+  ContextItem,
+  ContextProviderDescription,
+  ContextProviderExtras,
+  ContextSubmenuItem,
+  IContextProvider,
+  LoadSubmenuItemsArgs,
+} from "../../..";
+import { DB2_SELF_PROMPT, DB2_SYSTEM_PROMPT } from "./prompts";
+
+const db2ContextProviderDesc: ContextProviderDescription = {
+  title: "db2i",
+  displayTitle: "Db2i",
+  description: "Db2 for i Context Provider",
+  type: "normal",
+};
+
+/**
+ * - Get Existing Db2 connection from vscode
+ * - Use connection to build query for database to add additional context
+ *
+ */
+export class db2ContextProvider implements IContextProvider {
+  get description(): ContextProviderDescription {
+    return db2ContextProviderDesc;
+  }
+
+  getCurrentJob(): JobInfo {
+    const currentJob: JobInfo = JobManager.getSelection();
+    return currentJob;
+  }
+
+  private getDefaultSchema = (): string => {
+    const currentJob: JobInfo = this.getCurrentJob();
+    return currentJob?.job.options.libraries[0] || "QGPL";
+  };
+
+  async getSelfCodes(selected: JobInfo): Promise<SelfCodeNode[] | undefined> {
+    const current_job = selected.job.id;
+    const content = `
+    SELECT 
+      job_name, 
+      user_name, 
+      reason_code, 
+      logged_time, 
+      logged_sqlstate, 
+      logged_sqlcode, 
+      matches, 
+      stmttext, 
+      message_text, 
+      message_second_level_text,
+      program_library, 
+      program_name, 
+      program_type, 
+      module_name, 
+      client_applname, 
+      client_programid, 
+      initial_stack
+    FROM 
+      qsys2.sql_error_log, 
+      LATERAL (
+        SELECT * 
+        FROM TABLE(SYSTOOLS.SQLCODE_INFO(logged_sqlcode))
+      )
+    WHERE 
+      job_name = '${current_job}'
+    ORDER BY 
+      logged_time DESC
+  `;
+
+    try {
+      const result = await selected.job
+        .query<SelfCodeNode>(content)
+        .execute(10000);
+      if (result.success) {
+        const data: SelfCodeNode[] = result.data.map((row) => ({
+          ...row,
+          INITIAL_STACK: JSON.parse(row.INITIAL_STACK as unknown as string),
```

Comment:
Be careful here. I've seen this `JSON.parse` fail because `row.INITIAL_STACK` is not valid JSON. I'd put this into a function: 

```
INITIAL_STACK: tryGetJson(row.INITIAL_STACK as unknown as string),
```

Reply:
if the parse fails, should we throw an error or just ignore that entry?